weave-js/src/components/Timestamp.tsx

```diff
@@ -22,6 +23,49 @@ type TimestampProps = {
   // By default, a "relative" timestamp will update automatically.
   // If you don't want this behavior you can set this prop to false.
   live?: boolean;
+
+  // If true, will omit the time when it's midnight (00:00)
+  dropTimeWhenDefault?: boolean;
+};
+
+// Format a time difference to a micro string (1h, 1d, 1w, etc.)
+const formatMicroTime = (then: moment.Moment): string => {
+  const now = moment();
+  const diffMs = now.diff(then);
```

Comment:
I got confused here by the use of micro and "ms" - I think you are using micro to describe the small string format (eg 16d) but I was also thinking microsecond resolution time string, which is what our started and ended timestamps for calls use. Maybe we should call the short form strings something else